cirq-core/cirq/ops/gate_features.py

```diff
@@ -31,38 +30,7 @@ def qubit_index_to_equivalence_group_key(self, index: int) -> int:
         return 0
 
 
-class SupportsOnEachGate(raw_types.Gate, metaclass=abc.ABCMeta):
```

Comment:
What's our backwards compatibility story for `SupportsOnEachGate` (and--in general--"behavior mixin" classes)? I admit it may be a bit much to care in this case; but user code could conceivably  

 - using isinstance checks against it
 - using this as a mixin to their variadic gates

Reply:
Yeah I was wondering about this too. If we want to take it through a full deprecation path such that SingleQubitGate still inherits from SupportsOnEachGate due to isinstance checks, that will require fixing up of any unit test that has a single qubit gate, which....

Unless we can make a special deprecated attribute that doesn't throw when testing, for use in these kinds of situations?

Reply:
@balopat do you have any thoughts on the backwards compatibility questions here? Depending on the level of backwards compatibility we want to offer, It could substantially affect the size of this project .

Reply:
If I re-add `SupportsOnEachGate` as an empty class here, then that'll fix the 2nd problem: `class MyCustomGate(gate_features.SupportsOnEachGate)` will still compile and Gate will provide the `on_each` functionality. Preserving isinstance behavior is a lot more challenging. Need to think about that in terms of deprecating TwoQubitGate and ThreeQubitGate as well.

Reply:
Did this, and also used a metaclass for backwards isinstance compatibility, which was approved in a similar deprecation here https://github.com/quantumlib/Cirq/pull/4207#issuecomment-868854219.